### DISCUSSION TITLE
globalExtends are not fetched using token

### DISCUSSION THREAD

**Discussion by n0rad** [renovatebot/renovate#18894] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.22.2

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

globalExtends fail to load due to 404.
Looking at the log the configuration resolve correctly the platform and the token before trying to resolve the globalExtends, but it's not fetched with those info.
I had to explicitly set the platform to the preset, which tend to tell that the already resolved config is not used to fetch the global extends.
I also tried to add hostRules as env vars but it does not help

### Relevant debug logs

<details><summary>Logs</summary>

```
...
DEBUG: Env config
       "config": {"hostRules": [], "binarySource": "docker", "token": "***********"}
DEBUG: Combined config
       "config": {
         "autodiscover": false,
         "detectHostRulesFromEnv": true,
         "dryRun": null,
         "extends": ["blablacar/renovate-config//ep/default.json5"],
         "globalExtends": ["github>blablacar/renovate-config//ep/global-config.json5"],
         "persistRepoData": true,
         "platform": "github",
         "printConfig": true,
         "hostRules": [],
         "binarySource": "docker",
         "token": "***********",
         "repositories": ["blablacar/test"]
       }
DEBUG: GET https://api.github.com/repos/blablacar/renovate-config/contents/ep/global-config.json5 = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=178)

```
</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
it's not a bug. 

we simply don't have hostrules at that phase, so the http requests can't use any authentication at that time.

so you can't use private presets for global extends.

**Reply by rarkins** [reply]
Is it fundamentally impossible though? I was also curious whether  this was github.com for both?

**Reply by viceice** [reply]
it's for all platforms. global extends are resolved before platform is initialized.

**Reply by rarkins** [reply]
I meant it sounded like their platform and preset were both on GitHub.com. Maybe we could get it to work for local presets only using the token?

**Reply by viceice** [reply]
maybe if we set the default platform hostrule much earlier?

**Reply by rarkins** [reply]
Or initialize the platform first? It wouldn't mean many params are excluded from globalExtends

**Reply by n0rad** [reply]
fully `github.com` in my case. setup so:
```
DEBUG: File config
       "config": {
         "autodiscover": false,
         "detectHostRulesFromEnv": true,
         "dryRun": null,
         "extends": ["github>blablacar/renovate-config//ep/default.json5"],
         "globalExtends": ["github>blablacar/renovate-config//ep/global-config.json5"],
         "persistRepoData": true,
         "platform": "github",
         "printConfig": true
       }
DEBUG: CLI config
       "config": {"repositories": ["blablacar/test"]}
DEBUG: Env config
       "config": {
         "binarySource": "docker",
         "token": "***********"
       }
```
